File: app/src/main/java/com/checkin/app/checkin/manager/fragment/ManagerTablesActivateFragment.java
package com.checkin.app.checkin.manager.fragment;

import android.content.Context;
import android.widget.Button;

import com.checkin.app.checkin.Misc.BaseFragment;
import com.checkin.app.checkin.R;

import butterknife.BindView;
import butterknife.OnClick;

public class ManagerTablesActivateFragment extends BaseFragment {

    @BindView(R.id.btn_live_order_activate)
    Button btnLiveOrderActivate;

    private LiveOrdersInteraction mListener;

    public ManagerTablesActivateFragment() {
    }

    public static ManagerTablesActivateFragment newInstance() {
        return new ManagerTablesActivateFragment();
    }

    @Override
    protected int getRootLayout() {
        return R.layout.fragment_manager_tables_activate;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof LiveOrdersInteraction) {
            mListener = (LiveOrdersInteraction) context;
        } else {
            throw new ClassCastException(context.toString()
                    + " must implement ManagerWorkActivity.LiveOrdersInteraction");
        }
    }

    @OnClick(R.id.btn_live_order_activate)
    public void onViewClicked() {
        mListener.setLiveOrdersActivation(true);
    }

    public interface LiveOrdersInteraction {
        void setLiveOrdersActivation(boolean isActivated);
    }
}

File: app/src/main/java/com/checkin/app/checkin/manager/fragments/ManagerTablesActivateFragment.kt
package com.checkin.app.checkin.manager.fragments

import android.widget.Button
import butterknife.BindView
import butterknife.OnClick
import com.checkin.app.checkin.R
import com.checkin.app.checkin.misc.fragments.BaseFragment
import com.checkin.app.checkin.utility.parentActivityDelegate

class ManagerTablesActivateFragment : BaseFragment() {
    override val rootLayout: Int = R.layout.fragment_manager_tables_activate

    @BindView(R.id.btn_live_order_activate)
    internal lateinit var btnLiveOrderActivate: Button

    private val mListener: LiveOrdersInteraction by parentActivityDelegate()

    @OnClick(R.id.btn_live_order_activate)
    fun onViewClicked() {
        mListener.setLiveOrdersActivation(true)
    }

    interface LiveOrdersInteraction {
        fun setLiveOrdersActivation(isActivated: Boolean)
    }

    companion object {
        fun newInstance(): ManagerTablesActivateFragment = ManagerTablesActivateFragment()
    }
}